src/codegen/riscv64/constants-riscv64.h

```diff
@@ -978,6 +978,36 @@ class InstructionGetters : public T {
     return imm6 << 26 >> 26;
   }
 
+  inline int RvcImm6UValue() const {
+    DCHECK(this->IsShortInstruction());
+    // | funct3 | nzimm[17] | rs1/rd | nzimm[16:12] | opcode |
+    //  15         12                 6            2        
+    int32_t imm6 = this->RvcImm6Value();
+    DCHECK(imm6 != 0);
+    return (uint32_t)imm6 << 12 >> 12;
+  }
+  
+  inline int RvcImm6Addi16spValue() const {
+    DCHECK(this->IsShortInstruction());
+    // | funct3 | nzimm[9] | 2 | nzimm[4|6|8:7|5] | opcode |
+    //  15         12           6                2        
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm6 = ((Bits & 0x1000) >> 3) | ((Bits & 0x40) >> 2) 
+                  | ((Bits & 0x20) << 1) | ((Bits & 0x18) << 4) 
+                  | ((Bits & 0x4) << 3);
+    DCHECK(imm6 != 0);
```

Comment:
Is it `imm6` or `imm10` extracted? If it is `imm6` extracted, then it needs to be scaled by 16 to produce `imm10`.

The logic may be clearer if you extract `imm6` first, then scale `imm6` to `imm10`, then sign-extend at bit-10.

Reply:
It is imm10 extracted, scaled by 4.

I thought there is no need to extract imm6 first, because the bits in the instruction are imm[9:4]. Exracting imm10 is more easier than imm6.

I will change the variable imm6 -> imm10.